spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java

```diff
@@ -156,7 +159,46 @@ public boolean isSparkContextInitialized() {
   }
 
   static JobProgressListener setupListeners(SparkContext context) {
-    JobProgressListener pl = new JobProgressListener(context.getConf());
+    JobProgressListener pl = new JobProgressListener(context.getConf()) {
+      @Override
+      public synchronized void onJobStart(SparkListenerJobStart jobStart) {
+        super.onJobStart(jobStart);
+        int jobId = jobStart.jobId();
+        String jobGroupId = jobStart.properties().getProperty("spark.jobGroup.id");
+        String jobUrl = getJobUrl(jobId);
+        String noteId = getNoteId(jobGroupId);
+        String paragraphId = getParagraphId(jobGroupId);
+        if (jobUrl != null && noteId != null && paragraphId != null) {
+          RemoteEventClientWrapper eventClient = ZeppelinContext.getEventClient();
+          Map<String, String> infos = new java.util.HashMap<>();
+          infos.put("jobUrl", jobUrl);
+          infos.put("label", "SPARK JOB");
+          if (eventClient != null) {
+            eventClient.onParaInfosReceived(noteId, paragraphId, infos);
+          }
+        }
+      }
+
+      private String getJobUrl(int jobId) {
+        String jobUrl = null;
+        if (sparkUrl != null) {
+          jobUrl = sparkUrl + "/jobs/job?id=" + jobId;
+        }
+        return jobUrl;
+      }
+
+      private String getNoteId(String jobgroupId) {
+        int indexOf = jobgroupId.indexOf("-");
+        int secondIndex = jobgroupId.indexOf("-", indexOf + 1);
+        return jobgroupId.substring(indexOf + 1, secondIndex);
+      }
+
+      private String getParagraphId(String jobgroupId) {
+        int indexOf = jobgroupId.indexOf("-");
+        int secondIndex = jobgroupId.indexOf("-", indexOf + 1);
+        return jobgroupId.substring(secondIndex + 1, jobgroupId.length());
+      }
+    };
```

Comment:
Would it be better to put `getNoteId` and `getParagraphId` to Utils ? So that the constructing and parsing logic are in the same class.